engine/engine.py

```diff
@@ -87,7 +87,33 @@ class NewEngineNodeTree(bpy.types.Operator):
     bl_label = "New Node Tree"
 
     def execute(self, context):
-        bpy.ops.node.new_node_tree(type="DreamTexturesNodeTree")
+        # TODO: Get the name of the default node tree from a config file
+        # TODO: type is deprecated https://docs.blender.org/api/current/bpy.types.NodeTree.html 
+        #       When testing the type of the resulting node tree using  bpy.data.node_groups['Dream Textures Node Tree'].type it is '' because of that
+        bpy.ops.node.new_node_tree(type="DreamTexturesNodeTree", name="Dream Textures Node Editor")
+        # Get the newly generated node tree
+        node_trees = []
+        for nt in bpy.data.node_groups:
+            # the correct identifier is bl_idname nowadays
+            if nt.bl_idname == 'DreamTexturesNodeTree' and 'Dream Textures Node Editor' in nt.name:
+                node_trees.append(nt)
+        if node_trees != []:
+            # assume that the newly created node tree is the last one in the list
+            node_tree = node_trees[-1]
+            node_sd = node_tree.nodes.new(type="dream_textures.node_stable_diffusion")
+            node_sd.location = (200, 200)
+            node_out = node_tree.nodes.new(type="NodeGroupOutput")
+            node_out.location = (400, 200)
+            node_tree.links.new(node_sd.outputs['Image'], node_out.inputs[''])
```

Comment:
I think this connects the image to the virtual socket on the Group Output. You might need to add an image socket to it first, then link to that socket.

Reply:
It worked without but of course it should reflect what the user would get if they did it themselves so adding a color socket first is the cleaner solution for sure 👍🏻